### DISCUSSION TITLE
Discussion about tests before doing stuff

### DISCUSSION THREAD

**Discussion by ShuffleWire** [cyberbotics/webots#5507] [Q&A]
Don't know where to ask it, so here I come : 

What does the file tests/api/controllers/motor_spinner/motor_spinner.c is supposed to do/test ? as a testbench, it doesn't even use the ts module :/

**Comment by ad-daniel** [comment]
it's used to just generate motor actions, it's part of the the supervisor reset test. The actual test takes place in `supervisor_reset_physics_hinge`

**Reply by ShuffleWire** [reply]
yes, just figured out myself at the same time, sorry ! 
that being said, could we rewrite it in a may to avoid using multiple controller all over the place ? could the supervisor not be enough ?

**Reply by ad-daniel** [reply]
Probably yes, but the difference is really minor. Knowing that a supervisor can reset the physics of a node that isn't itself is arguably more interesting as a test I suppose, it indirectly tests something else
